src/stringify.ts

```diff
@@ -26,159 +26,195 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-let BARE_KEY = /^[a-z0-9-_]+$/i
+let BARE_KEY = /^[a-z0-9-_]+$/i;
 
-type ExtendedType = ReturnType<typeof extendedTypeOf>
-function extendedTypeOf (obj: any) {
-	let type = typeof obj
-	if (type === 'object') {
-		if (Array.isArray(obj)) return 'array'
-		if (obj instanceof Date) return 'date'
-	}
+type ExtendedType = ReturnType<typeof extendedTypeOf>;
+function extendedTypeOf(obj: any) {
+  let type = typeof obj;
+  if (type === 'object') {
+    if (Array.isArray(obj)) return 'array';
+    if (obj instanceof Date) return 'date';
+  }
 
-	return type
+  return type;
 }
 
-function isArrayOfTables (obj: any[]) {
-	for (let i = 0; i < obj.length; i++) {
-		if (extendedTypeOf(obj[i]) !== 'object') return false
-	}
+function isArrayOfTables(obj: any[]) {
+  for (let i = 0; i < obj.length; i++) {
+    if (extendedTypeOf(obj[i]) !== 'object') return false;
+  }
 
-	return obj.length != 0
+  return obj.length != 0;
 }
 
-function formatString (s: string) {
-	return JSON.stringify(s).replace(/\x7f/g, '\\u007f')
+function formatString(s: string) {
+  // Use multi-line literal strings for strings containing newlines or long strings
+  if ((s.includes('\n') || s.length > 120) && !s.includes("'''")) {
+    // For long strings without newlines, try to break them at spaces
+    if (!s.includes('\n') && s.length > 120) {
```

Comment:
`includes` is ran twice looking for the same thing. Avoid running the same linear search twice.